include/mbedtls/aes.h

```diff
@@ -67,6 +67,26 @@
 /* MBEDTLS_ERR_AES_HW_ACCEL_FAILED is deprecated and should not be used. */
 #define MBEDTLS_ERR_AES_HW_ACCEL_FAILED                   -0x0025  /**< AES hardware accelerator failed. */
 
+#if defined( MBEDTLS_CHECK_PARAMS )
+#define MBEDTLS_AES_VALIDATE_RET( cond )  do{ if( !(cond)  ) {                 \
```

Comment:
This is going to be reproduced in every public header. To reduce code duplication and be prepared in case these macros change, I suggest the addition of a header `check_params.h` which offers the following:

```c
#define MBEDTLS_VALIDATE_RET( error, cond )                             \
    do                                                                  \
    {                                                                   \
        if( !(cond)  )                                                  \
        {                                                               \
            MBEDTLS_PARAM_FAILED( #cond,                                \
                                 __FILE__,                              \
                                 __LINE__ );                            \
            return( error );                                            \
        }                                                               \
   } while( 0 )

#define MBEDTLS_VALIDATE( error, cond )                                 \
    do                                                                  \
    {                                                                   \
        if( !(cond)  )                                                  \
        {                                                               \
            MBEDTLS_PARAM_FAILED( #cond,                                \
                                 __FILE__,                              \
                                 __LINE__ );                            \
            return;                                                     \
        }                                                               \
   } while( 0 )
```

Each individual compilation unit can then define dedicated validation macros, e.g. in `aes.h`:
```c
#define MBEDTLS_AES_VALIDATE_RET( cond ) \
    MBEDTLS_VALIDATE_RET( MBEDTLS_ERR_AES_BAD_INPUT_DATA, cond )

#define MBEDTLS_AES_VALIDATE( cond ) \
    MBEDTLS_VALIDATE( MBEDTLS_ERR_AES_BAD_INPUT_DATA, cond )
```

Reply:
I agree, except:
- I'd rather put this in `platform_util.h` because that's the place we already have for such share utilities.
- I'd document these macros as in internal (not part of the public API)
- the macros in each compilation unit (C file) should not start with `MBEDTLS_` or check-names will complain.

Reply:
Thank you @mpg for taking a look.
* Agreed, `platform_util.h` is a good place; no need for a new header.
* Yes.
* Good point, I didn't think of that.

Reply:
@k-stachowiak Could you change this, please? As far as I see, it's the only point where the PR as-is deviates from the proposed internal interface. Everything else is implementation-detail as far as I see currently.